src/main/java/org/opentripplanner/gtfs/mapping/FareLegRuleMapper.java

```diff
@@ -23,12 +25,26 @@ public Collection<FareLegRule> map(
       .stream()
       .map(r -> {
         FareProduct productForRule = fareProductMapper.map(r.getFareProduct());
+        FareDistance fareDistance =
+          switch (r.getDistanceType()) {
+            case 0 -> new FareDistance.Stops(
+              r.getMinDistance().intValue(),
+              r.getMaxDistance().intValue()
+            );
+            case 1 -> new FareDistance.LinearDistance(
+              Distance.ofMeters(r.getMinDistance()),
+              Distance.ofMeters(r.getMaxDistance())
+            );
+            default -> null;
+          };
+
```

Comment:
This is not at the same abstraction level as the rest of the code here - extract into appropriate place.

Reply:
I think what Thomas means is that lines 28-40 should move into a method.